src/autosuggest/__tests__/autosuggest-input.test.tsx

```diff
@@ -0,0 +1,264 @@
+    const { wrapper } = render(
+      <AutosuggestInput
+        value="1"
+        onChange={() => undefined}
+        dropdownContent={
+          <div>
+            <button>content click</button>
+          </div>
+        }
+        dropdownFooter={<div>footer</div>}
+      />
+    );
+    findInput(wrapper).keydown(KeyCode.down);
+    expect(findOpenDropdown(wrapper)).not.toBe(null);
+    expect(getFocusables(wrapper.getElement()).length).toBeGreaterThan(2);
+  });
+
+  test('traps drodpown focus when footer is interactive', () => {
+    const { wrapper } = render(
+      <AutosuggestInput
+        value="1"
+        onChange={() => undefined}
+        dropdownContent={<div>content</div>}
+        dropdownFooter={
+          <div>
+            <button>footer click</button>
+          </div>
+        }
+      />
+    );
+    findInput(wrapper).keydown(KeyCode.down);
+    expect(findOpenDropdown(wrapper)).not.toBe(null);
+    expect(getFocusables(wrapper.getElement()).length).toBeGreaterThan(2);
+  });
+});
+
+describe('input events', () => {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  test('opens dropdown and calls onChange when input changes', () => {
+    const onChange = jest.fn();
+    const { wrapper } = render(<AutosuggestInput value="1" onChange={onChange} />);
+    expect(findOpenDropdown(wrapper)).toBe(null);
+    act(() => {
+      Simulate.change(findInput(wrapper).getElement(), { target: { value: '2' } as unknown as EventTarget });
+    });
+    expect(findOpenDropdown(wrapper)).not.toBe(null);
+    expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ detail: { value: '2' } }));
+  });
+
+  test('opens dropdown and calls onFocus when input receives focus', () => {
+    const onFocus = jest.fn();
+    const { wrapper } = render(<AutosuggestInput value="1" onChange={() => undefined} onFocus={onFocus} />);
+    expect(findOpenDropdown(wrapper)).toBe(null);
+    findInput(wrapper).focus();
+    expect(findOpenDropdown(wrapper)).not.toBe(null);
+    expect(onFocus).toHaveBeenCalled();
+  });
+
+  test('opens dropdown and calls onFocus when input loses focus', () => {
+    const onBlur = jest.fn();
+    const { wrapper } = render(<AutosuggestInput value="1" onChange={() => undefined} onBlur={onBlur} />);
+    expect(findOpenDropdown(wrapper)).toBe(null);
+    findInput(wrapper).focus();
+    findInput(wrapper).blur();
+    expect(findOpenDropdown(wrapper)).toBe(null);
+    expect(onBlur).toHaveBeenCalled();
+  });
+
+  test('calls onDelayedInput after small interval following the input value change', () => {
+    jest.useFakeTimers();
+    const onDelayedInput = jest.fn();
+    const { wrapper } = render(
+      <AutosuggestInput value="1" onChange={() => undefined} onDelayedInput={onDelayedInput} />
+    );
+    act(() => {
+      Simulate.change(findInput(wrapper).getElement(), { target: { value: '2' } as unknown as EventTarget });
+    });
```

Comment:
Any reason why you are writing it by hand instead of our util, which does the same: https://github.com/cloudscape-design/components/blob/75a6fdb641d49818c94fd156788756c346694b4e/src/test-utils/dom/input/base-input.ts#L35-L37

If you could not use it, because component is private, you can instantiate an instance of `BaseInputWrapper` directly

Reply:
Ah, I haven't thought about that -- thanks, will update 👍